### PR TITLE
Feat/progress bars

### LINKED ISSUE #446: Progress bars remake
Design: [zeplin](https://app.zeplin.io/project/5bcd9657767a220982a4ab61/screen/5bd8629015d2791b64732429)
Design docs: https://xx13.github.io/vuestic_redesign/#progress_bars

- [x] Remove vertical progress bar.

### VaProgressBar
Props:
- [x] `value` - number
- [x] `theme` - same as before
- [x] `indeterminate` - boolean, progress bar will just animate by itself.
- [x] `buffer` - number, second value, useful for preloading video etc.
image

`thin` and `thick` are low priority (you can consider ticket done without those). Split them into separate ticket if you don't plan to implement them.
- [ ] `thin` - boolean
- [ ] `thick` - boolean

Slots
- [x] `default`, that's the main text. We will not display any text if nothing is provided in this slot. (before we did show value in percents)

### VaProgressCircle
Props:
- [x] `value` - number
- [x] `theme` - same as before
- [x] `indeterminate` - boolean, progress circle will just animate by itself.

Slots
- [x] `default`, that's the main text. We will not display any text if nothing is provided in this slot. (before we did show value in percents)

Behaviour for circle and bar progress components:
- [x] If value goes past 100 - progress circle should stop at 100. No error is to be shown.
- [x] If value goes below 0 - progress circle should stop at 0. No error is to be shown.
- [x] component should be always animated, for instance, if you change value from 0 to 20 it should be done gradually. Consult references to figure optimal speed.

## Presentation & docs
- [x] Make a presentation page similarly to what is done for references (just visual stuff).
- [ ] Update docs

## Finalization
- [ ] [Component checks](https://github.com/epicmaxco/vuestic-admin/issues/378)

## References
* [Ant design](https://vuecomponent.github.io/ant-design-vue/components/progress/)
* [Vuetify](https://vuetifyjs.com/en/components/progress)
* [Quasar](https://vuecomponent.github.io/ant-design-vue/components/progress/)

image
image

### PR DESCRIPTION
**Pull request by ilearnjs**
close #446

### CODE REVIEW

File: src/components/statistics/progress-bars/Widgets/CircleBars.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <vuestic-widget class="progress-bar-widget" :headerText="$t('progressBars.circle')">
+    <div class="va-row">
+      <div v-for="n in 10" :key="n" class="flex sm1 md1">
+        <vuestic-circle-progress :value="value * n / 10"/>
+      </div>
+      <div class="flex sm1 md1">
+        <vuestic-circle-progress :indeterminate="true"/>
```
Comment (jean-moldovan) [inline src/components/statistics/progress-bars/Widgets/CircleBars.vue]: Same as `<vuestic-circle-progress indeterminate />`

File: src/components/statistics/progress-bars/Widgets/CircleBars.vue [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  data () {
+    return {
+      value: 0
+    }
+  },
+  mounted () {
+    this.animateValue()
+  },
+  methods: {
+    animateValue () {
```
Comment (jean-moldovan) [inline src/components/statistics/progress-bars/Widgets/CircleBars.vue]: Why would we need to make it go 0 -> 100 after timeout? How does animation work?

File: src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/CircleProgressBar.demo.vue [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
+    </VbContainer>
+
     <VbContainer title="Default">
       <circle-bar :value="value"/>
     </VbContainer>
 
-    <VbContainer title="No value">
-      <circle-bar/>
+    <VbContainer title="Danger">
+      <circle-bar :value="value" :theme="'Danger'"/>
```
Comment (jean-moldovan) [inline src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/CircleProgressBar.demo.vue]: Same as `<circle-bar :value="value" theme="Danger"/>`

File: src/vuestic-theme/vuestic-sass/resources/_variables.scss [hunk 4]
```diff
@@ -287,15 +287,11 @@ $vuestic-switch-padding: 0.313rem 2.375rem;
...
 
 //Progress Bars
-$progress-bar-value-font-size: 0.6875rem;
-$progress-bar-circle-diameter: 3.125rem;
-$progress-bar-circle-bw: .0678rem;
-$progress-bar-circle-overlay-diameter: calc(#{$progress-bar-circle-diameter} - 2 * #{$progress-bar-circle-bw});
-$progress-bar-vertical-height: 5.75rem;
+$progress-bar-value-font-size: 0.625rem;
+$progress-bar-circle-diameter: 2.5rem;
 $progress-bar-width-thin: .125rem;
```
Comment (asvae) [inline src/vuestic-theme/vuestic-sass/resources/_variables.scss]: Remove?

File: src/components/statistics/progress-bars/Widgets/CircleBars.vue (hunk 2 above)
Comment (jean-moldovan) [reply src/components/statistics/progress-bars/Widgets/CircleBars.vue]: ok

File: src/components/statistics/progress-bars/Widgets/CircleBars.vue (hunk 1 above)
Comment (ilearnjs) [reply src/components/statistics/progress-bars/Widgets/CircleBars.vue]: +

File: src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/CircleProgressBar.demo.vue (hunk 3 above)
Comment (ilearnjs) [reply src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/CircleProgressBar.demo.vue]: \+

File: src/components/statistics/progress-bars/Widgets/CircleBars.vue [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      <div class="flex sm1 md1">
+        <vuestic-circle-progress :indeterminate="true"/>
+      </div>
+    </div>
+    <div class="va-row">
+      <div v-for="n in 10" :key="n" class="flex sm1 md1">
+        <vuestic-circle-progress :value="value * n / 10" :theme="'Info'">{{ value * n / 10 }}%</vuestic-circle-progress>
+      </div>
+      <div class="flex sm1 md1">
+        <vuestic-circle-progress :indeterminate="true" :theme="'Info'"/>
```
[the remark this answers is not in the document]
Comment (ilearnjs) [reply src/components/statistics/progress-bars/Widgets/CircleBars.vue]: /+

File: src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/HorizontalProgressBar.demo.vue [hunk 6]
```diff
@@ -1,26 +1,62 @@
...
+    'horizontal-progress-bar': HorizontalProressBar
   },
+  data () {
+    return {
+      value: 35
+    }
+  }
 }
 </script>
+
```
[the remark this answers is not in the document]
Comment (ilearnjs) [reply src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/HorizontalProgressBar.demo.vue]: \+

File: src/vuestic-theme/vuestic-sass/resources/_variables.scss (hunk 4 above)
Comment (ilearnjs) [reply src/vuestic-theme/vuestic-sass/resources/_variables.scss]: \+

File: src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/HorizontalProgressBar.vue [hunk 7]
```diff
@@ -1,132 +1,163 @@
...
       type: String,
-      default: 'Primary',
-    },
-    size: {
-      type: String,
-      default: 'basic',
+      default: 'Primary'
     },
-    disabled: {
+    // If 'indeterminate' is 'true' 'value' and 'buffer' props will be ignored.
```
[the remark this answers is not in the document]
Comment (ilearnjs) [reply src/vuestic-theme/vuestic-components/vuestic-progress-bar/progress-types/HorizontalProgressBar.vue]: Please review  
https://github.com/epicmaxco/vuestic-admin/pull/458/commits/4167ce52b0215444fdfcce406d4145ba599162df

**Pull request merged**